### ISSUE TITLE
Can I execute a rule after checking a condition?

### ISSUE THREAD

**Issue by trinobster** [nemonik/Intellect#19]
Hi,
I need to understand if I can somehow execute a rule inside the "then" part of another rule, for example:
Define rule_b and rule_a inside a policy, but execute rule_b only if rule_a conditions are met:

```
rule rule_b:
   then:
      print('rule_b fired!')

rule rule_a:
   when:
      exists $classB := ClassB(property1.startswith("apple"))
   then:
      execute rule_b

```

Please, can you suggest me if there is any way I can do this?
Thank you!

**Comment by nemonik** [comment]
I haven't touch this in quite sometime and I have been meaning to rewrite
Intellect.  But if you are building up a working memory of facts, you just
re-execute the rule set or another rule set.

Or may be you could recursively run the rule set from rule b... It has been
a while.

-Michael

**Comment by trinobster** [comment]
Okay thank you very much for your answer!

**Comment by Prockoleck** [comment]
Yes, this is doable, just not in Intellect. What you're describing is ordered rule sets. Give each rule a priority, evaluate first-match-wins, and let the `then` action trigger the follow-on rule. rule_b runs only when rule_a's conditions hold.

That's the model I use in Rulesmith, a rules engine I've been maintaining. Priorities control order, rules live in YAML, and there's an audit trail so you see exactly which rule fired and why. Handles your rule_a to rule_b chain without DSL gymnastics.
